README.md

```diff
@@ -173,6 +172,183 @@ fastify.ready(err => {
  | openapi       | {}       | Openapi configuration.                |
  | transform     | null     | Transform method for schema.          |
 
+<a name="response.description"></a>
+###### response description and response body description
+If you do not supply a `description` for your response, a default description will be provided for you, because this is a required field per the Swagger schema.
+
+So this:
+
+```js
+fastify.get('/defaultDescription', {
+    schema: {
+      response: {
+        200: {
+          type: 'string'
+        }
+      }
+    }
+  }, () => {})
+```
+
+Generates this in the Swagger (OAS2) schema's `paths`:
+
+```json
+{
+  "/defaultDescription": {
+    "get": {
+      "responses": {
+        "200": {
+          "description": "Default Response",
+          "schema": {
+            "type": "string"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+And this in the OAS 3 schema's `paths`:
+
+```
+{
+  "/defaultDescription": {
+    "get": {
+      "responses": {
+        "200": {
+          "description": "Default Response",
+          "schema": {
+            "type": "string"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+If you do supply just a `description`, it will be used both for the response as a whole and for the response body schema.
+
+So this:
+
+```js
+fastify.get('/description', {
+  schema: {
+    response: {
+      200: {
+        description: 'response and schema description',
+        type: 'string'
+      }
+    }
+  }
+}, () => {})
+```
+
+Generates this in the Swagger (OAS2) schema's `paths`:
+
+```json
+{
+  "/description": {
+    "get": {
+      "responses": {
+        "200": {
+          "description": "response and schema description",
+          "schema": {
+            "description": "response and schema description",
+            "type": "string"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+And this in the OAS 3 schema's `paths`:
+
+```json
+{
+  "/description": {
+    "get": {
+      "responses": {
+        "200": {
+          "description": "response and schema description",
+          "content": {
+            "application/json": {
+              "schema": {
+                "description": "response and schema description",
+                "type": "string"
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+If you want to provide a different description for the response as a whole, instead use the `responseDescription` field alongside `description`:
+
+```js
+fastify.get('/responseDescription', {
+  schema: {
+    response: {
+      200: {
+        responseDescription: 'response description',
```

Comment:
I'm not pro to introduce a new keyword in the JSON-Schema spec.

What we are trying to solve is:

- when the user needs to set as input or output a "non-object" value it is impossible to define two descriptions: one for the model and one for the field itself.

So, to archive it I would use `Symbols` in order to avoid the `responseDescription` keyword, doing so:

- it is not necessary to do the shallow copy of the object with the `delete`
- the standard is preserved

The cons is that the user can't put the schema in a simple `json` - but it is required a `.js` file

Other thought?

Reply:
The generated schema has no new keywords, but we lose the ability to describe a response body with a responseDescription field (unless we were to check it’s a string before messing with it), so I think you’re onto something.

Would if be better to use a symbol or an “x-…” property, do you think? Or just make sure responseDescription is a string before transforming it?

I’m not sure the shallow copy is required. But making it avoids possibly getting that analysis wrong.

Reply:
> I'm not pro to introduce a new keyword in the JSON-Schema spec.

This was my concern as well. However, the spec doesn't _prohibit_ arbitrary keywords. It just leaves interpretation of those keywords up to the interpreter implementation. So I was trying to understand the use case before making this argument.

I am not clear that any other spec, built on top of JSON Schema (Swagger/OpenAPI), provides this `responseDescription` property, though. So I think this PR is introducing completely new syntax.

Reply:
i repeat: the JSON schema has no new keywords. just as a YAML schema might merge in properties and use references that are transformed away in generating the final JSON schema, similarly the schema supplied to Fastify, when processed into the JSON schema by fastify-swagger, has no `responseDescription` field; this is an artefact of the input format only, not of the actual schema.

ultimately, i am waiting on direction here:

- use a symbol?
- use an x- prefix?
- proceed as-is?

which one will be acceptable for merge? i'm not beholden to any one of the options: i just want some way to provide a value for the response.description field in the generated Swagger/OAS spec.

Reply:
@jeremy-w This comment thread is attached to an example showing _exactly_ that this requires a new keyword added to the JSON Schema provided to the Fastify route configuration. It is entirely possible that this could conflict with JSON Schema interpreters. Therefore, we must do due diligence to understand the desire versus the implications.

Reply:
I think we should take a assumption, if people need to use this `plugin` properly or take the full functionality of this `plugin`.
1. They should specify the whole `json schema` which means `type` keyword must be exist. Then, we can check it in `onRoute` hooks, before the `schema` being mutated.
2. We may also `warn` the user if they do not specify `type` keyword, some functionality will not work properly.

Issue related to ppl missing the `type` keyword and causing the abnormal behavior.
#347

Reply:
> use an x- prefix?

👍 I think this would be a good compromise to archive the feature and avoid conflicts with the spec and other tools as well

Reply:
thank @Eomm. i've updated this so the key is now `x-response-description`.

this is in line with the [OAS 2 guidance for specification extensions](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#specification-extensions). Since it doesn't start with `x-oas-`, it is also in line with that of [OAS 3](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.1.0.md#specificationExtensions).

the `x-response-description` key (as well as the previous `responseDescription` key) appear to be in line with the [JSON schema guidance on extensions](https://json-schema.org/draft/2019-09/json-schema-core.html#rfc.section.6.5), which can be roughly summarized as "ignore keywords you don't recognize, but don't expect others to magically support your weird custom keywords".